website/src/form-context.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { createContext, useContext, useState } from "react"
+import {
+  unstable_FormStateReturn as FormStateReturn,
+  unstable_useFormState as useFormState,
+} from "reakit/Form"
+import {
+  AnnotatedFormUpdate,
+  FormFieldsFragment,
+  useUpdateWordMutation,
+} from "./graphql/dailp"
+
+type FormContextType = {
+  form: FormStateReturn<any | undefined>
+  isEditing: boolean
+  setIsEditing: (bool: boolean) => void
+}
+
+const FormContext = createContext<FormContextType>({} as FormContextType)
+
+/** Instantiates a form state used to keep track of the current word and information about all its features. */
+export const FormProvider = (props: { children: any }) => {
+  const [isEditing, setIsEditing] = useState(false)
+  const word: FormFieldsFragment | undefined = undefined
+
+  const [updateWordResult, updateWord] = useUpdateWordMutation()
+
+  /** Calls the backend GraphQL mutation to update a word. */
+  const runUpdate = (variables: { word: AnnotatedFormUpdate }) => {
+    updateWord(variables).then((result) => {
+      console.log(result.data)
+    })
+  }
+
+  const form = useFormState({
+    values: {
+      word,
+    },
+    onValidate: (values) => {},
+    onSubmit: (values) => {
+      if (!values.word) {
+        throw { values: { word: "No word found" } }
+      }
```

Comment:
You can also check for `values.word` existing and being valid in the `onValidate` function.